doc/source/guide/in_depth/numpy_guide.rst

```diff
@@ -0,0 +1,124 @@
+.. _numpy_in_depth:
+
+##############################
+Using ODL with NumPy and SciPy
+##############################
+
+`Numpy
+<http://www.numpy.org/>`_ is the ubiquitous library for array computations in Python, and is used by almost all major numerical packages. It provides optimized `Array objects
+<http://docs.scipy.org/doc/numpy/reference/arrays.html>`_ that allow efficient storage of large arrays. It also provides several optimized algorithms for many of the functions used in numerical programming, such as taking the cosine or adding two arrays.
+
+`SciPy
+<http://www.scipy.org/>`_ is a library built on top of NumPy providing more advanced algorithms such as linear solvers, statistics, signal and image processing etc.
+
+Many operations are more naturally performed using NumPy/SciPy than with ODL and with that in mind ODL has been designed so that interfacing with them is as easy and fast as possible.
+
+Casting vectors to-from arrays
+------------------------------
+ODL vectors are stored in an abstract way, enabling storage on the CPU, GPU, or perhaps on a cluster on the other side of the world. This allows algorithms to be written in a generalized manner. Still, it is often convenient to be able to access the data and look at it, perhaps to initialize a vector, or to call an external function.
+
+To cast a NumPy array to a ODL vector, you simply need to call the `LinearSpace.element` method in an appropriate space::
+
+   >>> r3 = odl.Rn(3)
+   >>> arr = np.array([1, 2, 3])
+   >>> x = r3.element(arr)
+
+If the data type and storage methods allow it, the vector simply wraps the underlying array using a `view
+<http://docs.scipy.org/doc/numpy/glossary.html#term-view>`_::
+   
+   >>> float_arr = np.array([1.0, 2.0, 3.0])
+   >>> x = r3.element(float_arr)
+   >>> x.data is float_arr
+   True
+
+Casting ODL vectors to NumPy arrays can be done in two ways, either through the member function `NtuplesBaseVector.asarray`, or using `numpy.asarray`. These are both optimized and if possible return a view::
+
+   >>> x.asarray()
+   array([1.0, 2.0, 3.0])
+   >>> np.asarray(x)
+   array([1.0, 2.0, 3.0])
+
+These methods work with any ODL vector represented by an array, for example in discretizations a 2 dimensional array can be used::
+
+   >>> X = odl.uniform_discr([0, 0], [1, 1], [3, 3])
+   >>> arr = np.array([[1, 2, 3],
+   ...                 [4, 5, 6],
+   ...                 [7, 8, 9]])
+   >>> x = X.element(arr)
+   >>> x.asarray()
+   array([[1.0, 2.0, 3.0],
+          [4.0, 5.0, 6.0],
+          [7.0, 8.0, 9.0]])
+
+Using ODL vectors with NumPy functions
+--------------------------------------
+It is often useful to use NumPy functions with ODL vectors. For universal functions (`ufuncs
```

Comment:
A bit repetitious. Suggestion: "A very convenient feature of ODL is its seamless interaction with NumPy functions."